LibCrypto: Don't return reference to stack frame in PBKDF

BenWiederhake opened the pull request:
A reference to the current stack frame becomes invalid after returning, so returning Bytes is pointless.

I don't understand why this wasn't discovered earlier, but [it caused some CI problems for me](https://github.com/SerenityOS/serenity/pull/19209#issuecomment-1572599912), so I fixed it.

Don't take this as encouragement to break master! :^)

GMTA commented on Userland/Libraries/LibCrypto/Hash/PBKDF2.h:
```diff
@@ -77,7 +77,7 @@ class PBKDF2 {
 
         // 5. Output the derived key DK
-        return key;
+        return move(key);
```
A `move` for a return value should almost never be necessary

BenWiederhake replied:
And that's also what CI complains about. You're right, thanks :)

BenWiederhake commented:
Changes since last push:
- Don't `return move()` – I thought it was necessary, but apparently I didn't fully grasp the semantics of `move` around `return`.
- Rebased for convenience.
